src/classes/character/character.ts

```diff
@@ -0,0 +1,128 @@
+import _ from "lodash";
+import { Document } from "mongoose";
+import { ICharacter } from "../../types/character/character";
+import { IStat } from "../../types/character/stat";
+import { IItem } from "../../types/shop/item";
+import { IEffect } from "../../types/skill/effect";
+
+export class Character extends Document implements ICharacter {
+  public credits: ICharacter["credits"];
+  public effects?: ICharacter["effects"];
+  public experience: ICharacter["experience"];
+  public inventory?: ICharacter["inventory"];
+  public inventoryMax: ICharacter["inventoryMax"];
+  public inventorySpace: ICharacter["inventorySpace"];
+  public level: ICharacter["level"];
+  public name: ICharacter["name"];
+  public ownerId: ICharacter["ownerId"];
+  public stats: ICharacter["stats"];
+
+  public constructor() {
+    super();
+    this.credits = 300;
```

Comment:
Do you remember the other time when I said you did not have a need for a constructor?
That was false due to method usage.
But here you are simply setting primitives values so keep only the super and move all this setters directly with the variable declarations to reduce the code complexity.

Reply:
Oh ok. I will move those values.